.github/workflows/publish-to-test-pypi.yml

```diff
@@ -1,25 +1,39 @@
-name: Publish to PyPI
+# This workflow will upload a Python Package using Twine when a release is created
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python#publishing-to-package-registries
 
-on: create
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+name: Publish release to PyPI
+
+on:
+  release:
+    types: [published]
+
+permissions:
+  contents: read
 
 jobs:
- publish_to_pypi:
+  deploy:
+
     runs-on: ubuntu-latest
+
     steps:
-    - name: Check out repository
-      uses: actions/checkout@v2
+    - uses: actions/checkout@v4
     - name: Set up Python
-      uses: actions/setup-python@v2
+      uses: actions/setup-python@v3
       with:
-        python-version: '3.10'
+        python-version: '3.x'
```

Comment:
which versions does 3.x do? Is it all "current" versions or something?

Reply:
I'm guessing so, any non major change